Detections/MultipleDataSources/MalformedUserAgents.yaml

```diff
@@ -30,13 +33,23 @@ query: |
   (OfficeActivity | where TimeGenerated >= ago(endtime) | where UserAgent != ""),
   (OfficeActivity
   | where TimeGenerated >= ago(endtime)
-  | where RecordType in ("AzureActiveDirectoryAccountLogon", "AzureActiveDirectoryStsLogon")
-  | extend OperationName = Operation, UserAgent = extractjson("$[0].Value", ExtendedProperties, typeof(string))),
+  | where RecordType in ("AzureActiveDirectory", "AzureActiveDirectoryStsLogon")
+  | extend OperationName = Operation
+  | parse ExtendedProperties with * 'User-Agent\\":\\"' UserAgent2 '\\' *
+  | parse ExtendedProperties with * 'UserAgent",      "Value": "' UserAgent1 '"' *
+  | where isnotempty(UserAgent1) or isnotempty(UserAgent2)
+  | extend UserAgent = iff( RecordType == 'AzureActiveDirectoryStsLogon', UserAgent1, UserAgent2)
+  ),
   (AzureDiagnostics
   | where TimeGenerated >= ago(endtime)
   | where ResourceType =~ "APPLICATIONGATEWAYS" 
   | where OperationName =~ "ApplicationGatewayAccess" 
-  | extend ClientIP = columnifexists("clientIP_s", "None"), UserAgent = columnifexists("userAgent_s", "None")))
+  | extend ClientIP = columnifexists("clientIP_s", "None"), UserAgent = columnifexists("userAgent_s", "None")
+  ),
+  (SigninLogs
+  | where TimeGenerated >= ago(endtime)
```

Comment:
There are cases with SigninLogs events don't have UserAgent popluated, we should exclude rows right away where UserAgent is empty.

Reply:
Done now. Thanks Shain.